clang/lib/DPCT/Schema.h

```diff
@@ -0,0 +1,158 @@
+//===--------------------------- Schema.h ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef DPCT_SCHEMA_H
+#define DPCT_SCHEMA_H
+
+#include "AnalysisInfo.h"
+#include "llvm/Support/Casting.h"
+#include "llvm/Support/JSON.h"
+#include "llvm/Support/raw_ostream.h"
+#include <clang/AST/ASTContext.h>
+#include <clang/AST/Decl.h>
+#include <clang/AST/DeclCXX.h>
+#include <clang/Basic/LLVM.h>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace clang {
+namespace dpct {
+
+enum ValType : int {
+  ScalarValue = 0,
+  ArrayValue,
+  Pointer,
+  PointerOfPointer,
+};
+
+inline std::ostream &operator<<(std::ostream &strm, ValType tt) {
+  const std::string VTStr[] = {"Scalar", "Array", "Pointer",
```

Comment:
Should we use `const char * VTStr[]` to avoid construct a `std::string` array when this operator used.